### DISCUSSION TITLE
Letting a user update his own account details without visiting tenant.fief.dev with /api/profile or request to admin API

### DISCUSSION THREAD

**Discussion by pcnoic** [fief-dev/fief#298] [Q&A] [answered]
Hey folks, 

I am working on a Python integration with FastAPI and using Fief as my authentication and user management system. I would like to let the users of my app handle their account details from within my own frontend, without them having to visit <tenant>.fief.dev. I expected this to be possible from the `/api/profile` endpoint when sending a `PATCH` request.

I am using the user's `access_token` to forward the request from my own API to fief's tenant scoped api (<tenant>.fief.dev/api). I've setup two custom fields for my users, namely `name` and `company_name`. Here is the request body that I am sending:

`PATCH`
```
{
    "fields": {
        "name": "Christos Vasileios Alexiou",
        "company_name": "DalekDyne, Inc"
    }
}
```

The tenant API simply returns back user's details as they are right now:

```
{
    "sub": "5600d3d9-fc16-476e-83ad-redacted",
    "email": "redacted@gmail.com",
    "email_verified": true,
    "is_active": true,
    "tenant_id": "b07d5e37-830e-4b93-9d17-redacted",
    "fields": {
        "company": "Dalekdyne, Inc",
        "name": "Christos Vasileios Alexiou"
    }
}
```

I've also noticed that even if I set an empty JSON object to the API `/api/profile` as request body I get back the same answer.

I monitored the network calls from the default fief tenant management UI and I saw that the UI is achieving what I am trying to achieve using the `admin` api instead of the tenant api. Is this the only way to achieve it? 

![Screenshot 2023-11-05 at 4 56 32 PM](https://github.com/fief-dev/fief/assets/12615106/a1dd0651-cf2e-4fed-a4ec-568808ffbe52)

**Comment by frankie567** [comment, marked as answer]
I can confirm that the `PATCH /api/profile` endpoint is working. Make sure to use the right field key when submitting the payload. In the snipped you provided, you set `company_name` while the API returned you `company`.

**Comment by pcnoic** [comment]
This is indeed true Francois. I managed to get it working a short while
after opening this discussion but failed to report back.

Thanks for your time and help!
